atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableRendererV2.java

```diff
@@ -0,0 +1,422 @@
+                    .addModifiers(Modifier.PUBLIC);
+
+            getterBuilder = addParametersFromRowComponents(getterBuilder, tableMetadata);
+
+            getterBuilder.returns(ParameterizedTypeName.get(
+                    ClassName.get(Optional.class),
+                    ClassName.get(col.getValue().getValueType().getTypeClass())));
+            getterBuilder
+                    .addStatement("$T row = $T.of($N)", rowType, rowType, getArgumentsFromRowComponents(tableMetadata))
+                    .addStatement("byte[] bytes = row.persistToBytes()")
+                    .addStatement("$T colSelection = $T.create($T.singletonList($T.toCachedBytes($S)))",
+                            ColumnSelection.class, ColumnSelection.class, Collections.class,
+                            PtBytes.class, col.getShortName())
+                    .addStatement("$T<byte[]> rowResult = t.getRows(tableRef, $T.of(bytes), colSelection).get(bytes)",
+                            RowResult.class, ImmutableSet.class)
+                    .beginControlFlow("if (rowResult == null)")
+                        .addStatement("return $T.empty()", Optional.class)
+                    .endControlFlow()
+                    .beginControlFlow("else")
+                        .addStatement("return $T.of($T.of(rowResult).get$N())",
+                                Optional.class, rowResultType, VarName(col))
+                    .endControlFlow();
+
+            return getterBuilder.build();
+        }
+
+        private MethodSpec renderNamedGetSeveralRowsColumn(NamedColumnDescription col) {
+            Preconditions.checkArgument(tableMetadata.getRowMetadata().getRowParts().size() == 1);
+
+            NameComponentDescription rowComponent = tableMetadata.getRowMetadata().getRowParts().get(0);
+            MethodSpec.Builder getterBuilder = MethodSpec.methodBuilder("get" + VarName(col) + 's')
+                    .addModifiers(Modifier.PUBLIC)
+                    .addParameter(
+                            ParameterizedTypeName.get(
+                                    ClassName.get(Iterable.class),
+                                    ClassName.get(rowComponent.getType().getTypeClass())),
+                            "rowKeys");
+
+            getterBuilder.returns(ParameterizedTypeName.get(
+                    ClassName.get(List.class),
+                    ClassName.get(col.getValue().getValueType().getTypeClass())));
+
+            getterBuilder
+                    .addStatement("$T colSelection = $T.create($T.singletonList($T.toCachedBytes($S)))",
+                            ColumnSelection.class, ColumnSelection.class, Collections.class,
+                            PtBytes.class, col.getShortName())
+                    .addStatement("$T<$T> rows = $T\n"
+                                    + ".newArrayList(rowKeys)\n"
+                                    + ".stream()\n"
```

Comment:
hmm, does JavaPoet not have a better way of doing newlines?

Reply:
Their .addStatement also add semicolons.